### PR TITLE
feat: Add how-to guide for using config-drive

### PR DESCRIPTION
**Pull request by NamrataSitlani**

### CODE REVIEW

File: docs/howto/openstack/nova/config-drive.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
+# Store metadata on a configuration drive
+
+Configuration drive is a read only drive that is attached into an
```
Comment (foadlind) [inline docs/howto/openstack/nova/config-drive.md:3]: attached to ...

File: docs/howto/openstack/nova/config-drive.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
+# Store metadata on a configuration drive
+
+Configuration drive is a read only drive that is attached into an 
+instance during boot. The instance can then mount the drive and
+read files from it. Configuration drives are used as a data source for cloud-init
+and for also passing data to your instances to configure.
```
Comment (foadlind) [inline docs/howto/openstack/nova/config-drive.md:6]: and also for ...

File: docs/howto/openstack/nova/config-drive.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+--image ubuntu \
+--flavor m1.small \
+--key-name [KEYPAIR] \
+--nic net-id=[NETWORK_ID] \ 
+guest0
+
+```
+
+This command creates the config drive with volume label `config-2` which is
+attached to the instance during the boot and add the contents of the file `myscript.sh`
```
Comment (foadlind) [inline docs/howto/openstack/nova/config-drive.md:45]: and adds the contents ...

**Review by mrtmm — APPROVED** [review]
Looks good to me Namrata, thanks!

File: docs/howto/openstack/nova/config-drive.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
+# Store metadata on a configuration drive
+
+Configuration drive is a read only drive that is attached to an 
+instance during boot. The instance can then mount the drive and
+read files from it. Configuration drives are used as a data source for cloud-init
```
Comment (fghaas) [inline docs/howto/openstack/nova/config-drive.md:5]: This needs an external link to the cloud-init documentation.

File: docs/howto/openstack/nova/config-drive.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
+# Store metadata on a configuration drive
```
Comment (fghaas) [inline docs/howto/openstack/nova/config-drive.md:1]: Here we need an introductory paragraph that explains what problem this how-to solves.

File: docs/howto/openstack/nova/config-drive.md [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+Configuration drive is a read only drive that is attached to an 
+instance during boot. The instance can then mount the drive and
+read files from it. Configuration drives are used as a data source for cloud-init
+and also for passing data to your instances to configure.
+
+# Enable the configuration drive
+
+To enable the configuration drive, you need to pass the parameter `--config-drive`
+with `true` flag to the `openstack server create` command and can also
```
Comment (fghaas) [inline docs/howto/openstack/nova/config-drive.md:11]: This is a confusing run-on sentence, which will need to be broken in two.

File: docs/howto/openstack/nova/config-drive.md [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+specify the user data you want to inject into the server.
+
+## Example shell script
+
+A bash script is a common way to inject user data to an instance.
+The following script will install `tree` command during instance
```
Comment (fghaas) [inline docs/howto/openstack/nova/config-drive.md:17]: It's "*the* `tree` command". But `apt` does not install a command; it installs a package.

File: docs/howto/openstack/nova/config-drive.md [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+creation.
+
+#### **`myscript.sh`**
+```bash
+#!/bin/sh
+
+# Example script to run at boot
+# This installs tree
+sudo apt-get update
```
Comment (fghaas) [inline docs/howto/openstack/nova/config-drive.md:26]: Please don't use a shell script example that runs `sudo apt-get <something>`. That's not a good way to install packages. The way you do that is with

```yaml
#cloud-config
package_update: true
packages:
  - tree
 ```

File: docs/howto/openstack/nova/config-drive.md [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+--nic net-id=[NETWORK_ID] \ 
+guest0
+
+```
+
+This command creates the config drive with volume label `config-2` which is
+attached to the instance during the boot and adds the contents of the file `myscript.sh`
+to the `user_data` file in `openstack/{version}` directory.
+
+To access the config drive, we need to mount the config drive using the following instructions:
```
Comment (fghaas) [inline docs/howto/openstack/nova/config-drive.md:48]: Why is this here?

**Review by fghaas — CHANGES_REQUESTED** [review]
Thanks @NamrataSitlani, this is a good start but will need some rewrites.

File: docs/howto/openstack/nova/config-drive.md (hunk 9 above)
Comment (NamrataSitlani) [reply docs/howto/openstack/nova/config-drive.md:48]: I do not understand this comment. Can you please elaborate?

File: docs/howto/openstack/nova/config-drive.md (hunk 5 above)
Comment (NamrataSitlani) [reply docs/howto/openstack/nova/config-drive.md:27]: Done

File: docs/howto/openstack/nova/config-drive.md (hunk 6 above)
Comment (NamrataSitlani) [reply docs/howto/openstack/nova/config-drive.md]: Done

File: docs/howto/openstack/nova/config-drive.md (hunk 7 above)
Comment (NamrataSitlani) [reply docs/howto/openstack/nova/config-drive.md]: Done

File: docs/howto/openstack/nova/config-drive.md (hunk 9 above)
Comment (NamrataSitlani) [reply docs/howto/openstack/nova/config-drive.md]: I have removed the bit which explains how to access the config drive.
